### ISSUE TITLE
 Assertion failed at "/deps/zig/src/stage1/codegen.cpp:168" in get_llvm_cc, when building vulkan application with glfw for windows

### ISSUE THREAD

**Issue by SamuelDeboni** [ziglang/zig#7337]
I get this message when building my vulkan example for windows, on linux works fine.

```
Semantic Analysis [896/1566] Assertion failed at /deps/zig/src/stage1/codegen.cpp:168 in get_llvm_cc. This is a bug in the Zig compiler.
Unable to dump stack trace: debug info stripped
vulkan-exemple...The following command terminated unexpectedly:
/home/samuel/.local/opt/zigc/zig build-exe /home/samuel/dox/projects/vulkan-triangle/src/main.zig --library c --library glfw --library gdi32 --library shell32 --cache-dir /home/samuel/dox/projects/vulkan-triangle/zig-cache --name vulkan-exemple -target x86_64-windows-gnu -I /home/samuel/dox/projects/vulkan-triangle/include -L lib --enable-cache
error: the following build command failed with exit code 6:
/home/samuel/dox/projects/vulkan-triangle/zig-cache/o/128e4c1162f6a0bdcedb732e38dda2e8/build /home/samuel/.local/opt/zigc/zig /home/samuel/dox/projects/vulkan-triangle /home/samuel/dox/projects/vulkan-triangle/zig-cache -Dwindows
```

I narrowed down the error to this line, `vulkan_helper.zig:8`
```zig
pub extern "c" fn glfwGetInstanceProcAddress(instance: vk.Instance, procname: [*:0]const u8) vk.PfnVoidFunction;
```

[link to the project](https://gitlab.com/SamuelDeboni/zig-vulkan-example)

**Comment by LemonBoy** [comment]
Thanks for the bug report, the problem is a missing check in the compiler.
What the compiler is trying to tell you is that the definition of `vulkan_call_conv` is incorrect when targeting Windows, `.Stdcall` is only valid for 32bit targets.
The fix is to change `if (builtin.os.tag == .windows)` into `if (builtin.os.tag == .windows and builtin.arch == .i386)`.

One more detail, `extern "c"` means this function is externally defined and is provided by ... `libc`?
The value between the `""` has no effect beside adding `c` to the list of required libraries, it's working just fine here as it becomes a no-op given you already link against libc.

**Comment by andrewrk** [comment]
I wasn't able to cherry-pick the fix cleanly into 0.7.x so this fix will go into 0.8.0. I personally don't think it's worth the effort to make a 0.7.x-specific patch, but if someone decides they want it I won't turn down a patch.
